# Storing all markers type, and extended tf-tree
# Check YAML: http://www.yamllint.com/

marker_id:
  marker_0: {type: "tray_marker" }
  marker_1: {type: "tray_marker" }
  marker_2: {type: "tray_marker" }
  marker_3: {type: "tray_marker" }
  marker_100: {type: "compartment_marker" }
  marker_101: {type: "compartment_marker" }
  marker_102: {type: "compartment_marker" }
  marker_103: {type: "compartment_marker" }

marker_extended_frames:
  tray_marker:        [rescan_pos, pre_pick, insert, lift, post_pick]
  compartment_marker: [rescan_pos, pre_place, insert, drop, post_place]

markers_extended_tf:  # orien is opencv image convention
  tray_marker:
    rescan_pos: [0,   -0.10,   0.35,   1.57,   -1.57,    0]
    pre_pick:   [0,   -0.002,   0.33,   1.57,   -1.57,    0]
    insert:     [0,   -0.002,   0.023,  1.57,   -1.57,    0]
    lift:       [0,   -0.05,   0.023,  1.57,   -1.57,    0]
    post_pick:  [0,   -0.07,   0.42,   1.57,   -1.57,    0]
  compartment_marker:     
    rescan_pos: [0,   -0.10,   0.35,   1.57,   -1.57,    0]
    pre_place:  [0,   -0.2,   0.33,   1.57,   -1.57,    0]
    insert:     [0,   -0.2,   0,     1.57,   -1.57,    0]
    drop:       [0,   -0.15,  0,     1.57,   -1.57,    0]
    post_place: [0,   -0.15,  0.38,   1.57,   -1.57,    0]

# TODO
eef_to_tool_offset: 0.043
